### PR TITLE
[#10] egg.json for nest.land integration

### PR DESCRIPTION
**Pull request by tbaumer22**
# Issue

[#10](https://github.com/asos-craigmorten/opine/issues/10)

## Details

As we discussed in Discord, this PR includes an `egg.json` file, which is necessary for opine's listing on `nest.land`. Before this PR is merged, you'll want to visit our site to get an API key, download our CLI, and publish opine. Thanks for being open to this!

## CheckList

- [ ] PR starts with [_ISSUE_ID_] and branch name is e.g. feature/_description_ (see branching strategy in [CONTRIBUTING](./CONTRIBUTING.md)).
- [ ] Squashed commit message is prefixed with [_ISSUE_ID_] as above.
- [x] Has been tested (where required) before merge to develop.

### CODE REVIEW

File: egg.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "opine",
+    "description": "Fast, minimalist web framework for Deno ported from ExpressJS.",
+    "version": "0.10.0"
```
Comment (asos-craigmorten) [inline egg.json:4]:
```suggestion
    "version": "0.10.0",
```

**Comment by asos-craigmorten** [comment]
Uft that's annoying. TIL GH actions can't comment on PRs contributed from forks because it downgrades the permissions of the `GITHUB_TOKEN` it uses.

REF: https://github.community/t/github-actions-are-severely-limited-on-prs/18179

Will look into implementing on of the workarounds like [this one](https://github.com/nyurik/auto_pr_comments_from_forks) later maybe!

File: egg.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "opine",
+    "description": "Fast, minimalist web framework for Deno ported from ExpressJS.",
+    "version": "0.10.0",
+    "stable": true,
+    "files": [
+        "./mod.ts",
```
Comment (asos-craigmorten) [inline egg.json:7]:
```suggestion
        "./mod.ts",
        "./deps.ts",
        "./version.ts",
        "./lock.json",
```

File: egg.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        "./src/**/*",
+        "./README.md"
```
Comment (asos-craigmorten) [inline egg.json:9]:
```suggestion
        "./README.md",
        "./.github/CHANGELOG.md"
```

**Comment by asos-craigmorten** [comment]
@tbaumer22 do you know of anyone using a GitHub action for publishing packages to nest.land? Hoping to not have to do this manually, so guessing can probably do something like:

```yaml
name: Publish To Nest

on:
  release:
    types: [created]

jobs:
  publish-egg:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: denolib/setup-deno@master
        with:
          deno-version: 1.1.0
      - run: deno install -A -f --unstable -n eggs https://x.nest.land/eggs@0.1.0/mod.ts
      - run: make publishegg
        env:
          NEST_LAND_KEY: ${{secrets.NEST_LAND_KEY}}
```

Where `make publishegg` performs:

```bash
eggs link --key ${NEST_LAND_KEY}
eggs publish
```

I guess a "nice to have" would be a GitHub action provided by the `nestlandofficial` org so can just do something like:

```yaml
      - uses: nestlandofficial/publish
        env:
          KEY: ${{secrets.KEY}}
```

**Comment by asos-craigmorten** [comment]
Neverthess, I'll handle this in a separate PR - LGTM

**Comment by tbaumer22** [comment]
> @tbaumer22 do you know of anyone using a GitHub action for publishing packages to nest.land? Hoping to not have to do this manually, so guessing can probably do something like:
> 
> ```yaml
> name: Publish To Nest
> 
> on:
>   release:
>     types: [created]
> 
> jobs:
>   publish-egg:
>     runs-on: ubuntu-latest
>     steps:
>       - uses: actions/checkout@v2
>       - uses: denolib/setup-deno@master
>         with:
>           deno-version: 1.1.0
>       - run: deno install -A -f --unstable -n eggs https://x.nest.land/eggs@0.1.0/mod.ts
>       - run: make publishegg
>         env:
>           NEST_LAND_KEY: ${{secrets.NEST_LAND_KEY}}
> ```
> 
> Where `make publishegg` performs:
> 
> ```shell
> eggs link --key ${NEST_LAND_KEY}
> eggs publish
> ```
> 
> I guess a "nice to have" would be a GitHub action provided by the `nestlandofficial` org so can just do something like:
> 
> ```yaml
>       - uses: nestlandofficial/publish
>         env:
>           KEY: ${{secrets.KEY}}
> ```

This is a great idea. A template repository for this was in the pipeline. I guess you beat us to it! If you don't mind, we'll plan on using this as part of the template. (Credit will be provided!)

**Comment by asos-craigmorten** [comment]
@tbaumer22 yep feel free to use - final working version is here --> https://github.com/asos-craigmorten/opine/blob/main/.github/workflows/publish-egg.yml

And you can see it running and working here --> https://github.com/asos-craigmorten/opine/runs/783606309?check_suite_focus=true (assuming permissions allow)

**Pull request merged**